pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/MetsTransformerWorkerService.scala

```diff
@@ -33,25 +30,28 @@ class MetsTransformerWorkerService(
       processAndLog
     )
 
-  def processAndLog(metsData: MetsData): Future[Unit] =
-    process(metsData).recover {
+  def processAndLog(key: Version[String, Int]): Future[Unit] =
+    Future.fromTry(process(key).toTry).recover {
       case t =>
-        error(s"There was an error processing $metsData: ", t)
+        error(s"There was an error processing $key: ", t)
         throw t
     }
 
-  private def process(metsData: MetsData) = {
+  private def process(key: Version[String, Int]) = {
     for {
-      metsXmlInputStream <- fetchMets(metsData)
-      mets <- Future.fromTry(MetsXmlParser(metsXmlInputStream).toTry)
-      work <- Future.fromTry(mets.toWork(metsData.version).toTry)
-      _ <- Future.fromTry(messageSender.sendT(work))
+      metsString <- fetchMets(key)
+      mets <- MetsXmlParser(metsString)
+      work <- mets.toWork(key.version)
+      _ <- messageSender.sendT(work).toEither
     } yield ()
   }
 
-  private def fetchMets(metsData: MetsData) = {
-    Future {
-      s3Client.getObject(s3Config.bucketName, metsData.path).getObjectContent
+  private def fetchMets(key: Version[String, Int]) = {
+    store.get(key) match {
+      case Left(err)                   => Left(new Exception(err.e))
```

Comment:
Is there any need to wrap within a new exception? I think `err.e` is an exception itself?

Reply:
You're right, no need